### DISCUSSION TITLE
Update Bounds after Clip

### DISCUSSION THREAD

**Discussion by adam-grant-hendry** [pyvista/pyvista#3525] [Q&A] [answered]
After clipping an object, it seems the bounds are not updated. Is there a way to update them?

For example, I would have assumed the white outline below to fit exactly to the blue half-cylinder, but it still appears to go to the extents of the original tan cylinder.

```python
import pyvista as pv

plotter = pv.Plotter()
plotter.add_axes()

cylinder = pv.Cylinder()
cylinder.actor = plotter.add_mesh(cylinder, color='tan')

roi = pv.Cube(
    center=[0, 0, .5],
    x_length=2,
    y_length=2,
    z_length=1,
)

roi.actor = plotter.add_mesh(roi, color='red', opacity=0.5)

clipped = cylinder.clip_box(roi.bounds)
clipped.translate((2, 2, 2), inplace=True)
clipped.actor = plotter.add_mesh(clipped, color='blue')

bbox = clipped.outline()
bbox.actor = plotter.add_mesh(bbox)

plotter.show()
```

![image](https://user-images.githubusercontent.com/59346180/198901664-19947cf1-336d-4686-bcdd-8a5ee96f9859.png)

**NB:** _I'm rotated ~180-deg about z in the screenshot above, so when the window first pops up, the blue cylinder will be closer to you._

**Comment by banesullivan** [comment]
Specifically, which bounds?

**Reply by adam-grant-hendry** [reply]
Ya, my bad: this was a pretty bare-bones post. Allow me to do better. I've updated my question with an MRE.

**Reply by adam-grant-hendry** [reply]
Also, I added this printout as a double-check:

```python
cyl_xmin, cyl_xmax, cyl_ymin, cyl_ymax, cyl_zmin, cyl_zmax = cylinder.bounds
cyl_width = cyl_xmax - cyl_xmin
cyl_height = cyl_ymax - cyl_ymin
cyl_depth = cyl_zmax - cyl_zmin
cyl_geometry = (cyl_width, cyl_height, cyl_depth)

clip_xmin, clip_xmax, clip_ymin, clip_ymax, clip_zmin, clip_zmax = clipped.bounds
clip_width = clip_xmax - clip_xmin
clip_height = clip_ymax - clip_ymin
clip_depth = clip_zmax - clip_zmin
clip_geometry = (clip_width, clip_height, clip_depth)

print(f'Cylinder Geometry (tan): {cyl_width} X {cyl_height} X {cyl_depth}')
print(f'Clipped Bounds (blue): {clip_width} X {clip_height} X {clip_depth}')
print(f'Same Extents?: {clip_geometry == cyl_geometry}')
```

which resulted in

```bash
Cylinder Geometry (tan): 1.0 X 1.0 X 1.0
Clipped Bounds (blue): 1.0 X 1.0 X 1.0
Same Extents?: True
```

**Reply by adam-grant-hendry** [reply]
@banesullivan I was able to devise a workaround by using `extract_surface()`. However, I'm curious to know if the intended behavior of the `clip_box()` is to not update the bounding box?

**Reply by banesullivan** [reply]
I think this might be a VTK issue... I've seen this before with a few things....

**Reply by adam-grant-hendry** [reply]
Quite interesting...yes, I think this behavior is repeatable in ParaView as well...

**Comment by adam-grant-hendry** [comment, marked as answer]
As a (temporary) workaround, one can use `extract_surface()` on the clipped surface and generate the `outline()` for that:

```python
clipped = cylinder.clip_box(roi.bounds).extract_surface()
``` 

which gives the desired result:

![image](https://user-images.githubusercontent.com/59346180/198915801-85b5e307-58de-43cd-b485-cc61033a5d22.png)
